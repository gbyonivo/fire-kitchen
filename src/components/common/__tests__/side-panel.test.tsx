import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { SidePanel } from "../side-panel";

describe("SidePanel", () => {
  test.each([[true], [false]])("%s should match snapshot", async (open) => {
    const onClose = jest.fn();
    const { container } = render(
      <SidePanel title="Test" open={open} onClose={onClose} id="test-id">
        <div>Test</div>
      </SidePanel>
    );

    if (open) {
      await userEvent.hover(screen.getByTestId("test-id"));
    }
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
