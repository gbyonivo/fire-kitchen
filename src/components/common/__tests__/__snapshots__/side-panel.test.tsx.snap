// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel false should match snapshot 1`] = `<div />`;

exports[`SidePanel true should match snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
